general-superstaq/general_superstaq/superstaq_client.py

```diff
@@ -448,3 +450,25 @@ def __repr__(self) -> str:
                 verbose={self.verbose!r},
             )"""
         )
+
+
+def find_api_key() -> Optional[str]:
+    """Try to load a SuperstaQ API key from the environment or a key file."""
+
+    if os.getenv("SUPERSTAQ_API_KEY"):
+        return os.getenv("SUPERSTAQ_API_KEY")
+
+    data_dir = pathlib.Path(os.getenv("XDG_DATA_HOME", "~/.local/share")).expanduser()
+    home_dir = pathlib.Path.home()
+    for directory in [
+        data_dir.joinpath("super.tech"),
+        data_dir.joinpath("coldquanta"),
+        home_dir.joinpath(".super.tech"),
+        home_dir.joinpath(".coldquanta"),
+    ]:
+        path = directory.joinpath("superstaq_api_key")
+        if path.is_file():
+            with open(path, "r") as file:
+                return file.readline()
+
+    return None
```

Comment:
wdyt about raising the environment error here, rather than duplicating it in both clients?

Reply:
done